app/controllers/admin/project_logs_controller.rb

```diff
@@ -1,12 +1,51 @@
 class Admin::ProjectLogsController < Admin::AdminController
-  def destroy
+  include TranslationSaveable
+
+  def show
     @log = ProjectLog.find(params[:id])
-    @step = ProjectStep.find(@log.project_step_id)
-    @loan = Loan.find(@step.project_id)
+    authorize_with_parents
 
-    authorize @log
-    authorize @step
-    authorize @loan
+    redirect_to admin_loan_path(@loan)
+  end
+
+  def new
+    @log = ProjectLog.new(project_step_id: params[:step_id])
+    authorize_with_parents
+    @progress_metrics = ProjectLog.progress_metric_options
+    render "admin/logs/new", layout: false
+  end
+
+  def edit
+    @log = ProjectLog.find(params[:id])
+    authorize_with_parents
+    @progress_metrics = ProjectLog.progress_metric_options
+    render "admin/logs/edit", layout: false, locals: {log: @log}
+  end
+
+  def create
+    @log = ProjectLog.new(project_log_attribs)
+    authorize_with_parents
+    save_and_render_partial
+  end
+
+  def update
+    @log = ProjectLog.find(params[:id])
+    @log.assign_attributes(project_log_attribs)
+    authorize_with_parents
+
+    if params[:step_completed_on_date] == '1'
```

Comment:
Good job on this code!

QUESTION: Is there a reason this is only in update and not create?